src/algorithms/classic/graph_based/a_star.py

```diff
@@ -117,11 +118,16 @@ def f(self, x: Point) -> float:
         return ret
 
     def _follow_back_trace(self):
+        grid: Map = self._get_grid()
+        
         trace: List[Point] = self.get_back_trace(self._get_grid().goal)
         trace.reverse()
         for t in trace:
             self.move_agent(t)
             self.key_frame(ignore_key_frame_skip=True)
+        if isinstance(self._get_grid(), RosMap):
```

Comment:
You should send a waypoint between `self.move_agent` and `self.key_frame` so that the animation is in sync with the current position of the robot, which is how WPN does it.

Reply:
Yea that makes sense. I'll do that for both